src/views/SafeAreaView.js

```diff
@@ -0,0 +1,225 @@
+import React, { Component } from 'react';
+import {
+  DeviceInfo,
+  Dimensions,
+  InteractionManager,
+  NativeModules,
+  Platform,
+  SafeAreaView,
+  View,
+} from 'react-native';
+import withOrientation from './withOrientation';
+
+const { isIPhoneX_deprecated } = DeviceInfo;
+// See https://mydevice.io/devices/ for device dimensions
+const X_WIDTH = 375;
+const X_HEIGHT = 812;
+const PAD_WIDTH = 768;
+const PAD_HEIGHT = 1024;
+
+const { height: D_HEIGHT, width: D_WIDTH } = Dimensions.get('window');
+
+const { PlatformConstants = {} } = NativeModules;
+const { minor = 0 } = PlatformConstants.reactNativeVersion || {};
+
+const isIPhoneX = (() => {
+  if (minor >= 50) {
+    return isIPhoneX_deprecated;
+  }
+
+  return (
+    Platform.OS === 'ios' &&
+    ((D_HEIGHT === X_HEIGHT && D_WIDTH === X_WIDTH) ||
+      (D_HEIGHT === X_WIDTH && D_WIDTH === X_HEIGHT))
+  );
+})();
+
+const isIPad = (() => {
+  if (Platform.OS !== 'ios' || isIPhoneX) return false;
+
+  // if portrait and width is smaller than iPad width
+  if (D_HEIGHT > D_WIDTH && D_WIDTH < PAD_WIDTH) {
+    return false;
+  }
+
+  // if landscape and height is smaller that iPad height
+  if (D_WIDTH > D_HEIGHT && D_HEIGHT < PAD_WIDTH) {
+    return false;
+  }
+
+  return true;
+})();
+
+const statusBarHeight = isLandscape => {
```

Comment:
is this a reliable indicator of the status bar height, or do you mean to use https://github.com/facebook/react-native/pull/16478 when it's available?

Reply:
Funny, one of the comments on that PR says to rely on SafeAreaView ;-)

The idea right now is that SafeAreaView will eventually handle this in native-land and devs who use SafeAreaView will never have to worry about the status bar again. This is for sure a naive solution, but it is actually an improvement on what React Nav does right now. It does need to get better.